homeassistant/components/sleepiq/__init__.py

```diff
@@ -50,6 +59,23 @@ def setup(hass, config):
         return False
 
     hass.data[DOMAIN] = data
+
+    def handle_set_number(call):
+        bed = call.data.get(BED, "")
+        side = call.data.get(SIDE, "")
+        new_sleep_number = call.data.get(SLEEP_NUMBER, 0)
+
+        data.set_new_sleep_number(bed, side, new_sleep_number)
+
+    def handle_set_favorite(call):
+        bed = call.data.get(BED, "")
+        side = call.data.get(SIDE, "")
+
+        data.set_to_favorite_sleep_number(bed, side)
+
+    hass.services.register(DOMAIN, SERVICE_SET_SLEEP_NUMBER, handle_set_number)
+    hass.services.register(DOMAIN, SERVICE_SET_FAVORITE, handle_set_favorite)
```

Comment:
We already have binary sensors for these, right?
Why not register these services as entity services for those binary sensors?

<https://developers.home-assistant.io/docs/dev_101_services/#entity-services>

Reply:
I was trying to provide the minimal amount of changes to add the set number function.

Reply:
Loose integration services generally end up having way more code than entity services.
For example, you now have additional validation, need to find the bed and side (or the user has to pass them. While it could be something like:

```yaml
- service: sleepiq.set_sleep_number
  data:
    entity_id: binary_sensor.my_bed_side
    sleep_number: 60
```

So, let start with the first question: Do we already have binary sensors for each bed & side?